vm/sdk/src/call.rs

```diff
@@ -12,5 +20,25 @@ pub fn call(address: Address, input: Option<Vec<u8>>, method: Option<&str>, amou
   let input_payload = payload.encode();
   let input_len = input_payload.len();
   let input = crate::bytes_to_u32_vec(input_payload);
-  athena_vm::syscalls::call(address.as_ptr(), input.as_ptr(), input_len, amount.as_ptr());
+
+  // Allocate 1024 bytes on the heap, using u32 for proper alignment
+  let mut output_buf = Vec::<std::mem::MaybeUninit<u32>>::with_capacity(1024 / 4);
+
+  let output_size = athena_vm::syscalls::call(
+    address.as_ptr(),
+    input.as_ptr(),
+    input_len,
+    output_buf.as_mut_ptr() as *mut u32,
+    1024,
+    amount.as_ptr(),
+  );
+
+  // Convert the Vec<MaybeUninit<u32>> into a Vec<u8> without freeing buffer
+  let raw_ptr = output_buf.as_mut_ptr() as *mut u8;
+  let capacity = output_buf.capacity();
+  // Prevent `buffer` from being dropped, as we're transferring ownership
+  std::mem::forget(output_buf);
```

Comment:
consider this instead, since the intent is a bit clearer:

```
  // Prevent `buffer` from being dropped, as we're transferring ownership
  let mut output_buf = std::mem::ManuallyDrop::new(output_buf);
```

You could remove this line entirely, and add the new line above, before using `output_buf` after the sys call.